Analyzer/plugins/Analyzer.cc

```diff
@@ -3961,7 +3992,192 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+        //only consider good vertex
+      if (vertexColl[i].isFake() || fabs(vertexColl[i].z()) > 24 || vertexColl[i].position().rho() > 2 ||
+          vertexColl[i].ndof() <= 4) continue;
+      if(!firstpvfound) {
+          firstpvfound=true;
+          index_pv=i;
+      }
+    }
+
+  // loop on the tracks
+  bool apply_trigger_sel_for_calib = true;
+  if (typeMode_ != 3 && (!apply_trigger_sel_for_calib || (apply_trigger_sel_for_calib && trigInfo_>0)) && index_pv>-1) { 
+    for(unsigned int c=0;c<trackCollectionHandle->size();c++){
+      reco::TrackRef track = reco::TrackRef( trackCollectionHandle.product(), c );
+
+      // selection of the tracks
+
+      bool passedTrackCutsArray[8];
+      std::fill(std::begin(passedTrackCutsArray), std::end(passedTrackCutsArray),true);
+
+      // Check if eta is inside the max eta cut for detector homogeneity
+      passedTrackCutsArray[0]  = (fabs(track->eta()) < globalMaxEta_) ? true : false;
+      // Select only high purity tracks to ensure good quality tracks
+      passedTrackCutsArray[1]  = (track->quality(reco::TrackBase::highPurity)) ? true : false;
+      // Cut on the chi2 / ndof to ensure good quality tracks
+      passedTrackCutsArray[2] = ((track->chi2() / track->ndof()) < globalMaxChi2_) ? true : false;
+      // Check the min fraction of valid hits to ensure good stats on the hits
+      passedTrackCutsArray[3]  = (track->validFraction() > globalMinFOVH_) ? true : false;
+      // Cut on the impact parameter to ensure the track is coming from the PV
+      // for typeMode_ 3 (TOF only) dz and dxy is supposed to come from the beamspot
+      float dz = track->dz(vertexColl[index_pv].position());
+      float dxy = track->dxy(vertexColl[index_pv].position());
+      passedTrackCutsArray[4] = (fabs(dz) < 0.5) ? true : false;
+      passedTrackCutsArray[5] = (fabs(dxy) < 0.5)  ? true : false;
+
+      if (!passPreselection(passedTrackCutsArray)) continue;
+
+
+      //load dedx quantity associated to this track 
+      const reco::DeDxHitInfo* dedxHits = nullptr;
+      reco::DeDxHitInfoRef dedxHitsRef = dedxCollH->get(track.key());
+      if (!dedxHitsRef.isNull())
+        dedxHits = &(*dedxHitsRef);
+
+//      if(!dedxHits)  cout << "track " << c << " pt : " <<  track->pt()  << ", eta : " << track->eta()  << " without dedxHit info " << endl;
+      if(!dedxHits)continue;
+//      cout << "track " << c << " pt : " <<  track->pt()  << ", eta : " << track->eta()  << " with dedxHit info " << endl;
+
+      reco::DeDxHitInfoRef dedxHitsPrescRef;
+      dedxHitsPrescRef = dedxCollH->get(track.key());
+      int preScaleForDeDx = 1;
+      if (!dedxHitsPrescRef.isNull()) {
+        preScaleForDeDx = (*dedxPrescCollH)[dedxHitsPrescRef];
+      }
+//      if (preScaleForDeDx > 1) cout << "preScaleForDeDx is non-zero " << preScaleForDeDx << " for track pt " << track->pt() << endl;
+
+
+      float dEdxErr = 0;
+
+      // Ih no pixel L1
+      auto dedxIh_noL1_Tmp =
```

Comment:
Maybe this should be renamed so it's obvious that this is not the same `dedxIh_noL1_Tmp` as the one 2000 lines above? Also true for the other Ih and Ias

Reply:
It is the same definition. The first one is in the HSCP loop, the 2nd is in the track loop. So I prefer to use the same name.

Reply:
> The first one is in the HSCP loop, the 2nd is in the track loop.

exactly so it will to different values, no?

Reply:
It will give the same value for the tracks which are HSCP candidates above (because of same dEdX inputs). But here I run over all the general tracks, considering also low pT. There is no pT cut in my track loop. So at the end, if I would have plotted the histograms for dedxIh_noL1 (HSCP) or for dedxIh_noL1 (track), I would have seen two different histograms. I hope it is more clear.